controllers/dscinitialization/utils.go

```diff
@@ -426,6 +428,64 @@ func (r *DSCInitializationReconciler) createOdhCommonConfigMap(ctx context.Conte
 	return nil
 }
 
+func (r *DSCInitializationReconciler) createOdhTrustedCABundleConfigMap(ctx context.Context, namespace string, dscInit *dsci.DSCInitialization) error {
+	caConfigMapName := "odh-trusted-ca-bundle"
+	caDataFieldName := "odh-ca-bundle.crt"
+
+	// Expected configmap for the given namespace
+	desiredConfigMap := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      caConfigMapName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/part-of": "opendatahub-operator",
+				// Label required for the Cluster Network Operator(CNO) to inject the cluster trusted CA bundle
+				// into .data["ca-bundle.crt"]
+				"config.openshift.io/inject-trusted-cabundle": "true",
+			},
+		},
+		// Add the DSCInitialzation specified TrustedCABundle to the odh-ca-bundle.crt data field
+		// Additionally, the CNO operator will automatically create and maintain ca-bundle.crt
+		//  based on the application of the label 'config.openshift.io/inject-trusted-cabundle'
+		Data: map[string]string{caDataFieldName: dscInit.Spec.TrustedCABundle},
+	}
+
+	// Create Configmap if doesn't exist
+	foundConfigMap := &corev1.ConfigMap{}
+	err := r.Client.Get(ctx, client.ObjectKey{
+		Name:      caConfigMapName,
+		Namespace: namespace,
+	}, foundConfigMap)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			// Set Controller reference
+			err = ctrl.SetControllerReference(dscInit, foundConfigMap, r.Scheme)
```

Comment:
so we will need to have a DSCI CR in order to keep this configmap , in order make cert work?

Reply:
Yes, I think for any valid workflow of ODH operator,, we require DSCI instance to be created.

Alternatively, we would need a separate controller then if we want to watch over namespaces irrespective of DSC/DSCI resources. The reason I added this is DSCI controller is, since we would also want to update the configmap on changes in the `TrustedCABundle` field.

Let me know if this approach looks fine to you